charms/kserve-controller/src/charm.py

```diff
@@ -76,6 +97,30 @@ def _context(self):
             "cert": f"'{ca_context.decode('utf-8')}'",
         }
 
+    @property
+    def _inference_service_context(self):
+        """Context for rendering the inferenceservive-config ConfigMap."""
+        # Ensure any input is valid for deployment mode
+        deployment_mode = self.model.config["deployment-mode"].lower()
+        if deployment_mode == "serverless":
+            deployment_mode = "Serverless"
+        elif deployment_mode == "rawdeployment":
+            deployment_mode = "RawDeployment"
+        else:
+            self.unit.status = BlockedStatus(
+                "Please set deployment-mode to either Serverless or RawDeployment"
+            )
```

Comment:
should not we exit from here?

otherwise we proceed with the wrong depl mode

Reply:
I have updated this with a `return`. Let me know what you think.